lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -826,43 +945,35 @@ def __init__(
                 dictionary represents the Prometheus scrape
                 configuration for a single job. When not provided, a
                 default scrape configuration is provided for the
-                `/metrics` endpoint pooling all units of this charm on port `80`.
+                `/metrics` endpoint polling all units of the charm on port `80`
+                using the `MetricsEndpointProvider` object.
             alert_rules_path: an optional path for the location of alert rules
                 files.  Defaults to "./prometheus_alert_rules",
                 resolved from the directory hosting the charm entry file.
                 The alert rules are automatically update on charm upgrade.
         """
         if not relation_name:
-            # Check if there is just one relation with the right interface
-            prometheus_scrape_relations = [
-                relation_name
-                for relation_name, relation_meta in charm.meta.provides.items()
-                if relation_meta.interface_name == "prometheus_scrape"
-            ]
-
-            if len(prometheus_scrape_relations) == 1:
-                relation_name = prometheus_scrape_relations[0]
-            elif len(prometheus_scrape_relations) == 0:
+            relation_interface = RELATION_INTERFACE_NAME
+            try:
+                # Check if there is just one relation with the right interface
+                relation_name = _get_single_relation_by_interface(
+                    charm, relation_interface, RelationDirection.PROVIDED
+                )
+            except NoRelationWithInterfaceFoundError:
                 raise ModelError(
-                    "No provided relation with the 'prometheus_scrape' interface found"
+                    f"No provided relation with the '{relation_interface}' interface found; "
+                    f"did you add a relation with the '{relation_interface}' interface in the charm's "
+                    "metadata.yaml file?"
                 )
-            else:
+            except MultipleRelationsWithInterfaceFoundError as e:
                 raise ModelError(
-                    "Multiple relations with the 'prometheus_scrape' interface found: "
-                    f" {prometheus_scrape_relations}"
+                    f"Multiple provided relations with the '{relation_interface}' interface found: "
+                    f"{e.relations}; you must specify which relation should be managed by this "
+                    f"{type(self)} by providing the `relation_name` argument."
```

Comment:
I know it's annoying, especially this late, but we discussed not using Python features > 3.5. Especially if this library gets used by machine charms, it's easy to squash the f-strings now.